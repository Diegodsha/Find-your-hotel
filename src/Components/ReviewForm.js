import { useState } from 'react';
import PropTypes from 'prop-types';
import { StyledForm } from '../Assets/StyledSignFrom';

const ReviewForm = ({ addReview }) => {
  const [Review, setReview] = useState({
    reviewTitle: '',
    reviewBody: '',
    reviewScore: 0,
  });

  const handleChange = (e) => {
    setReview({
      ...Review,
      [e.target.name]: e.target.value,
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    addReview(Review);
    setReview({
      ...Review, reviewTitle: '', reviewBody: '', reviewScore: 0,
    });
  };

  return (
    <div>
      <button
        style={{ backgroundColor: '#1eac1e', border: 'none' }}
        type="button"
        className=" w-100 p-3 text-white fw-bold"
        data-bs-toggle="modal"
        data-bs-target="#exampleModal"
      >
        Add a review
      </button>

      <div
        className="modal fade"
        id="exampleModal"
        tabIndex="-1"
        aria-labelledby="exampleModalLabel"
        aria-hidden="true"
      >
        <div className="modal-dialog ">
          <div className="modal-content p-4">
            <div className="modal-header">
              <h5 className="modal-title text-center" id="exampleModalLabel">
                Thanks for your time, your opinion is very important for us!
              </h5>
            </div>
            <div className="modal-body ">
              <StyledForm
                id="reviewForm"
                className="d-flex flex-column align-items-center"
                onSubmit={handleSubmit}
              >
                <input
                  className="my-3 title userName"
                  placeholder="Title"
                  type="text"
                  value={Review.reviewTitle}
                  onChange={handleChange}
                  name="reviewTitle"
                  id="reviewTitle"
                />
                <textarea
                  className="my-3 body userName"
                  placeholder="Tell us what you think..."
                  value={Review.reviewBody}
                  onChange={handleChange}
                  name="reviewBody"
                  id="reviewBody"
                  minLength="4"
                  maxLength="240"
                />
                <input
                  className="my-3 score userName"
                  type="number"
                  value={Review.reviewScore}
                  onChange={handleChange}
                  placeholder="score"
                  name="reviewScore"
                  id="reviewScore"
                  min="1"
                  max="5"
                />
                <button type="submit" className="btn btn-primary sign-btn">
                  Add review
                </button>
              </StyledForm>
            </div>
            <div className="modal-footer d-flex justify-content-center">
              <button
                type="button"
                className="btn btn-secondary rounded-pill"
                data-bs-dismiss="modal"
              >
                Close
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

ReviewForm.propTypes = {
  addReview: PropTypes.func.isRequired,
};

export default ReviewForm;
