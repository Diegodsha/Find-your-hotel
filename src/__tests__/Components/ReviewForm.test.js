import renderer from 'react-test-renderer';
import ReviewForm from '../../Components/ReviewForm';

describe('ReviewForm Component', () => {
  const addReview = () => {};
  test('Renders correctly component', () => {
    const tree = renderer.create(<ReviewForm addReview={addReview} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
