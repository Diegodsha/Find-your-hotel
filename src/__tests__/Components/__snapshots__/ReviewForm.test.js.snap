// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewForm Component Renders correctly component 1`] = `
<div>
  <button
    className=" w-100 p-3 text-white fw-bold"
    data-bs-target="#exampleModal"
    data-bs-toggle="modal"
    style={
      Object {
        "backgroundColor": "#1eac1e",
        "border": "none",
      }
    }
    type="button"
  >
    Add a review
  </button>
  <div
    aria-hidden="true"
    aria-labelledby="exampleModalLabel"
    className="modal fade"
    id="exampleModal"
    tabIndex="-1"
  >
    <div
      className="modal-dialog "
    >
      <div
        className="modal-content p-4"
      >
        <div
          className="modal-header"
        >
          <h5
            className="modal-title text-center"
            id="exampleModalLabel"
          >
            Thanks for your time, your opinion is very important for us!
          </h5>
        </div>
        <div
          className="modal-body "
        >
          <form
            className="sc-gtsrHT gUeXta d-flex flex-column align-items-center"
            id="reviewForm"
            onSubmit={[Function]}
          >
            <input
              className="my-3 title userName"
              id="reviewTitle"
              name="reviewTitle"
              onChange={[Function]}
              placeholder="Title"
              type="text"
              value=""
            />
            <textarea
              className="my-3 body userName"
              id="reviewBody"
              maxLength="240"
              minLength="4"
              name="reviewBody"
              onChange={[Function]}
              placeholder="Tell us what you think..."
              value=""
            />
            <input
              className="my-3 score userName"
              id="reviewScore"
              max="5"
              min="1"
              name="reviewScore"
              onChange={[Function]}
              placeholder="score"
              type="number"
              value={0}
            />
            <button
              className="btn btn-primary sign-btn"
              type="submit"
            >
              Add review
            </button>
          </form>
        </div>
        <div
          className="modal-footer d-flex justify-content-center"
        >
          <button
            className="btn btn-secondary rounded-pill"
            data-bs-dismiss="modal"
            type="button"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
